feat(User) : 이메일을 이용한 회원 가입 API

seyoung5744 opened the pull request:
## 관련 이슈
- #14 

<br>

## 변경사항

- 이메일, 비밀번호 정규식을 검사합니다.
- 이미 가입된 유저일 경우 "이미 가입된 회원입니다." 라는 에러를 반환합니다.
- 회원의 status가 "탈퇴" 일 경우 회원가입 시 status만 active로 변경하여 재 가입합니다.
- 비밀 번호의 경우 암호화를 하여 DB에 저장합니다.

<br>


## 체크 리스트 (Checklist)

pull request 전에 아래 체크 리스트들을 만족하는 지 확인한 후 체크('x') 표시를 해주시기 바랍니다.

- [x]  master 브랜치가 아니라 **develop** 브랜치에 Merge하도록 pull request를 작성 중이신가요?
- [x]  모든 **테스트**가 성공했나요?

seyoung5744 committed:
feat(User) : 이메일을 이용한 회원 가입 API

- 이메일, 비밀번호 정규식을 검사합니다.
- 이미 가입된 유저일 경우 "이미 가입된 회원입니다." 라는 에러를 반환합니다.
- 회원의 status가 "탈퇴" 일 경우 회원가입 시 status만 active로 변경하여 재 가입합니다.
- 비밀 번호의 경우 암호화를 하여 DB에 저장합니다.

paik-dongjoo commented on src/main/java/com/zerobase/wishmarket/domain/user/model/dto/SignUpForm.java:
```diff
@@ -0,0 +1,29 @@
+@AllArgsConstructor
+public class SignUpForm {
+    @NotNull
+    @Email(message = "옳바른 이메일 형식이 아닙니다.")
```
간단한 내용이지만 오타인 것 같습니다.
"올바른 이메일 형식이 아닙니다." 로 수정하면 좋을 것 같아요.

seyoung5744 replied:
하하하...수정할게요

paik-dongjoo commented on src/main/java/com/zerobase/wishmarket/domain/user/model/dto/UserDto.java:
```diff
@@ -0,0 +1,29 @@
+@Getter
+@Builder
+@ToString
+@AllArgsConstructor
```
현재 UserDto 에서 선언되어 있는 변수들과 회원가입시 받아야 하는 필수 입력값 데이터들에는 차이가 있는 것처럼, 실제 Dto를 사용할 때는 @AllArgsConstructor 로만 구현하기 곤란한 상황도 있을 것 같은데 @Builder 어노테이션을 적용하는 건 어떨지 질문드립니다!

seyoung5744 replied:
UseDto를 공통적으로 사용될거라 예상해서  @AllArgsConstructor로 설정했습니다.
근데 회원 가입 시에서 따로 응답하는 DTO로 명칭을 바꿔서 사용해야할 거 같네요

Ko-udon commented on src/main/java/com/zerobase/wishmarket/domain/user/service/UserSignUpService.java:
```diff
@@ -0,0 +1,84 @@
+    }
+
+    private boolean checkInvalidPassword(String password) {
+        return !Pattern.matches("^(?=.*[A-Za-z])(?=.*\\d)(?=.*[$@$!%*#?&])[A-Za-z\\d$@$!%*#?&]{8,}$",password);
```
해당 정규식표현이 [최소 8자 + 최소 한개의 영문자 + 최소 한개의 숫자 + 최소 한개의 특수 문자] 가 맞을까요?
확인차 궁금해서 물어봅니다!

seyoung5744 replied:
넵 이해하신게 맞습니다

Ko-udon commented on src/test/java/com/zerobase/wishmarket/domain/user/service/UserSignUpServiceTest.java:
```diff
@@ -0,0 +1,112 @@
+    private UserSignUpService userSignUpService;
+
+
+    @Test
```
올바르지 않은 이메일 형태의 SignUp 테스트도 있으면 좋을 것 같습니다!

seyoung5744 committed:
fix(user) : 이메일, 비밀번호 정규식 수정

- FE 와 BE 의 이메일 , 비밀번호 정규식을 맞춰 진행합니다.

seyoung5744 committed:
fix(user) : 옳->올 맞춤법 수정

seyoung5744 committed:
fix(user) : UserDto를 회원가입 시에만 사용되도록 SignUpEmailDto로 수정

- UserDto라는 애매한 명칭 SignUpEmailDto로 수정
- Entity에서 SignUpEmailDto로 builder하는 것을 SignUpEmailDto에서 entity를 받아서 budilder 하는 것으로 수정

seyoung5744 committed:
Merge branch 'dev' into USER-REGISTER-API
